examples/ogbn_papers_100m.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+import os
+import os.path as osp
+import time
+
+import torch
+import torch.nn.functional as F
+from ogb.nodeproppred import PygNodePropPredDataset
+from torchmetrics import Accuracy
+
+import torch_geometric
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import GCNConv
+from torch_geometric.utils import to_undirected
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--hidden_channels', type=int, default=64)
+parser.add_argument('--lr', type=float, default=0.01)
+parser.add_argument('--epochs', type=int, default=3)
+parser.add_argument('--batch_size', type=int, default=128)
+parser.add_argument('--fan_out', type=int, default=50)
+
+args = parser.parse_args()
+
+if torch.cuda.is_available():
+    device = torch.device('cuda')
+else:
+    device = torch.device('cpu')
+
+dataset = PygNodePropPredDataset(name='ogbn-papers100M')
+split_idx = dataset.get_idx_split()
+data = dataset[0]
+data.y = data.y.reshape(-1).to(torch.long)
```

Comment:
I am pretty certain this explains your index issues. Can you drop this here (since `NaN` cannot be converted to `long`), and just do the conversion inside each mini-batch?